test/e2e/reconciler/cluster/controller_test.go

```diff
@@ -104,28 +107,37 @@ func TestClusterController(t *testing.T) {
 				}, wait.ForeverTestTimeout, time.Millisecond*100)
 
 				t.Logf("Expecting same spec to show up in sink")
-				cowboy.SetNamespace(targetNamespace)
-				err = servers[sinkClusterName].expect(cowboy, func(object *wildwestv1alpha1.Cowboy) error {
-					if diff := cmp.Diff(cowboy.Spec, object.Spec); diff != "" {
-						return fmt.Errorf("saw incorrect spec on sink cluster: %s", diff)
+				require.Eventually(t, func() bool {
+					if got, err := servers[sinkClusterName].client.Cowboys(testNamespace).Get(ctx, cowboy.Name, metav1.GetOptions{}); err != nil {
+						if apierrors.IsNotFound(err) {
+							return false
+						}
+						require.NoError(t, err, "Error getting cowboy %q in sink", cowboy.Name)
+
+						return false
+					} else if diff := cmp.Diff(cowboy.Spec, got.Spec); diff != "" {
+						require.Errorf(t, nil, "Spec mismatch on sink cluster: %s", diff)
```

Comment:
Suggest logging the diff (e.g. with t.Errorf) and returning false rather than failing on it.

Reply:
this one is fatal for the test IMO.